examples/load_and_reindex_bboxes.py

```diff
@@ -0,0 +1,336 @@
+# has no annotation defined.
+
+ds_ff = ds.reindex(
+    {"time": list(range(ds.time[-1].item()))},
+    method="ffill",  # propagate last valid index value forward
+)
+
+# check the first 14 frames of the data
+print("Position data array (first 14 frames):")
+print(ds_ff.position.data[:14, 0, :])  # time, individual, space
+
+print("----")
+print("Shape data array (first 14 frames):")
+print(ds_ff.shape.data[:14, 0, :])  # time, individual, space
+
+# %%
+# Fill in empty values with NaN
+# ----------------------------------------------------
+# Alternatively, we can fill in the missing frames with NaN values.
+# This can be useful if we want to interpolate the missing values later.
+ds_nan = ds.reindex(
+    {"time": list(range(ds.time[-1].item()))},
+    method=None,  # default
+)
+
+# check the first 14 frames of the data
+print("Position data array (first 14 frames):")
+print(ds_nan.position.data[:14, 0, :])
+
+print("----")
+print("Shape data array (first 14 frames):")
+print(ds_nan.shape.data[:14, 0, :])
+
+# %%
+# Linearly interpolate NaN values
+# ----------------------------------------------------------
+# We can instead fill in the missing values in the dataset applying linear
+# interpolation to the ``position`` and ``shape`` data arrays. In this way,
+# we would be assuming that the centroid of the bounding box moves linearly
+# between the two annotated values, and its width and height change linearly
+# as well.
+#
+# We use the dataset with NaN values as an input to the
+# ``interpolate_over_time`` function.
+ds_interp = ds_nan.copy()
+
+for data_array_str in ["position", "shape"]:
+    ds_interp[data_array_str] = interpolate_over_time(
+        data=ds_interp[data_array_str],
+        method="linear",
+        max_gap=None,
+        print_report=False,
+    )
+
+# check the first 14 frames of the data
+print("Position data array (first 14 frames):")
+print(ds_interp.position.data[:14, 0, :])
+
+print("----")
+print("Shape data array (first 14 frames):")
+print(ds_interp.shape.data[:14, 0, :])
+
+
+# %%
+# Compare interpolation methods
+# ------------------------------
+# We can now qualitatively compare the three different methods of filling
+# in the missing frames, by plotting the bounding boxes
+# for the first 6 frames of the video.
+#
+# Remember only frames 0 and 5 are annotated in the original dataset. These
+# are plotted in blue, while the forward filled values are plotted in orange
+# and the linearly interpolated values in green.
+
+# sphinx_gallery_thumbnail_number = 2
+
+# initialise figure
+fig = plt.figure(figsize=(8, 8))
+
+# loop over frames
+for frame_n in range(6):
+    # add subplot axes
+    ax = plt.subplot(3, 2, frame_n + 1)
+
+    # plot frame
+    # note: the video is indexed at every frame, so
+    # we use the frame number as index
+    ax.imshow(video[frame_n])
+
+    # plot bounding box for each dataset
+    for ds_i, ds_one in enumerate([ds_nan, ds_ff, ds_interp]):
```

Comment:
I was confues by the use of `ds_one`. I would expect `ds_` as a quasi-convention in this case.
`ds_one` made me think of "first dataset".

Reply:
I used `ds_filled` (`ds_` seemed to close to the loop variable `ds_i`)